### PR TITLE
contour: init at 0.1.1

### PR DESCRIPTION
**Pull request by moni-dz**
###### Motivation for this change

Testing notes:
 - doesn't work on wayland unless using the x11 backend (xwayland) with `QT_QPA_PLATFORM=xcb`, as it gives out `QWaylandGLContext: failed to create EGLContext, error=3009`

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/terminal-emulators/contour/default.nix [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+  buildInputs = [ freetype libGL pcre ];
+
+  meta = with lib; {
+    description = "Modern C++ Terminal Emulator";
+    homepage = "https://github.com/christianparpart/contour";
+    changelog = "https://github.com/christianparpart/contour/releases/tag/v${version}";
+    license = licenses.asl20;
+    maintainers = with maintainers; [ fortuneteller2k ];
+    platforms = platforms.linux ++ platforms.darwin;
```
Comment (SuperSandro2000) [inline pkgs/applications/terminal-emulators/contour/default.nix:25]:
```suggestion
    platforms = platforms.unix;
```

File: pkgs/applications/terminal-emulators/contour/default.nix [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  };
+
+  nativeBuildInputs = [ cmake pkg-config ];
+
+  buildInputs = [ freetype libGL pcre ];
+
+  meta = with lib; {
+    description = "Modern C++ Terminal Emulator";
+    homepage = "https://github.com/christianparpart/contour";
+    changelog = "https://github.com/christianparpart/contour/releases/tag/v${version}";
```
Comment (SuperSandro2000) [inline pkgs/applications/terminal-emulators/contour/default.nix:22]:
```suggestion
    changelog = "https://github.com/christianparpart/contour/blob/HEAD/Changelog.md";
```

File: pkgs/applications/terminal-emulators/contour/default.nix [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+mkDerivation rec {
+  pname = "contour";
+  version = "0.1.1";
+
+  src = fetchFromGitHub {
+    owner = "christianparpart";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "sha256-P7t+M75ZWjFcGWngcbaurdit6e+pb0ILljimhYqW0NI=";
+    fetchSubmodules = true;
```
Comment (SuperSandro2000) [inline pkgs/applications/terminal-emulators/contour/default.nix:12]:
```suggestion
```

there are no submodules.

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/applications/terminal-emulators/contour/default.nix (hunk 3 above)
Comment (fortuneteller2k) [reply pkgs/applications/terminal-emulators/contour/default.nix:12]: the v0.1.1 tag has submodules

**Pull request merged**